### PR TITLE
Improve Keyboard Shortcuts. Add alt-arrows & alt-shift-arrows

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/riot-web/issues/301

- [x] scroll to room tile so its visible when the room is active
- [x] expand room sublist when we enter it so we can see the room

Test build: https://riots.im/adhoc/shortcuts2/
The new test build (v2) no longer disables RLS' stickyRoom so expect that failure mode until https://github.com/vector-im/riot-web/issues/11743 is fixed.

### CODE REVIEW

**Comment by ara4n** [comment]
If you order rooms by unread, it's a bit surprising that as you flip through the rooms you mark them read... and so their position jumps to the read section of the list, so if you're trying to alt-up/down through the list one by one, it's a bit surprising that the list jumps around from under you.

Not sure what the right solution to that is, other than possibly debouncing the jumps somehow?

**Comment by t3chguy** [comment]
I believe that is what sticky room is meant to do, I disabled it in the test build as it's very very jumpy when switching successively

File: src/components/structures/RoomSubList.js [hunk 1]
```diff
@@ -111,17 +111,26 @@ export default class RoomSubList extends React.PureComponent {
...
+                if (this.props.list.some((r) => r.roomId === payload.roomId)) {
+                    this.forceUpdate();
+                }
+                break;
+
+            case 'view_room':
+                if (this.state.hidden && !this.props.forceExpand &&
+                    this.props.list.some((r) => r.roomId === payload.room_id)
+                ) {
+                    this.onClick();
```
Comment (dbkr) [inline src/components/structures/RoomSubList.js]: Would suggest splitting this out to a separate method that does whatever the `onClick` causes to happen, and then call that both from here and onClick which avoids the trap of someone adding the event parameter to `onClick` which will then NPE when called from here (and maybe make it more self-documenting as I can't see without going & reading the onClick why this is necessary).

File: src/components/views/rooms/RoomList.js [hunk 2]
```diff
@@ -242,6 +245,54 @@ export default createReactClass({
...
+                    ...historicalRooms,
+                    ...serverNoticeRooms,
+                ], this.props.searchFilter);
+
+                if (payload.unread) {
+                    // filter to only notification rooms (and our current active room so we can index properly)
+                    rooms = rooms.filter(room => {
+                        return room.roomId === currentRoomId || Unread.doesRoomHaveUnreadMessages(room);
+                    });
+                }
```
Comment (dbkr) [inline src/components/views/rooms/RoomList.js]: I'm surprised by how much this is having to implement its own version of a sequential list of the rooms, but I guess this is all being rewritten on Travis's branch anyway?

Comment (t3chguy) [reply src/components/views/rooms/RoomList.js]: I believe so, or if not it will be cleaned after the rewrite of the RLS

**Pull request merged**